docs/Reference/JSON-RPC-API-Methods.md

```diff
@@ -1929,6 +1929,168 @@ None
+
+**Parameters** 
+
+`data` - 20-byte address of proposed validator 
+
+**Returns** 
+
+`result: boolean` - `true`
+
+!!! example
+    ```bash tab="curl HTTP request"
+    $ curl -X POST --data '{"jsonrpc":"2.0","method":"ibft_discardValidatorVote","params":["0xef1bfb6a12794615c9b0b5a21e6741f01e570185"], "id":1}' <JSON-RPC-http-endpoint:port>
+    ```
+    
+    ```bash tab="wscat WS request"
+    {"jsonrpc":"2.0","method":"ibft_discardValidatorVote","params":["0xef1bfb6a12794615c9b0b5a21e6741f01e570185"], "id":1}
+    ```
+    
+    ```json tab="JSON result"
+    {
+      "jsonrpc" : "2.0",
+      "id" : 1,
+      "result" : true
+    }
+    ```
+
+### ibft_getPendingVotes
+
+Returns [current votes](../Consensus-Protocols/IBFT.md#adding-and-removing-validators). 
+
+**Parameters**
+
+None
+
+**Returns** 
+
+`result`: `object` - Map of account addresses to corresponding boolean values indicating the vote for each account. 
+
+If the boolean value is `true`, the vote is to add a validator. If `false`, the proposal is to remove a validator. 
+
+!!! example
+    ```bash tab="curl HTTP request"
+    $ curl -X POST --data '{"jsonrpc":"2.0","method":"ibft_getPendingVotes","params":[], "id":1}' <JSON-RPC-http-endpoint:port>
+    ```
+    
+    ```bash tab="wscat WS request"
+    {"jsonrpc":"2.0","method":"ibft_getPendingVotes","params":[], "id":1}
+    ```
+    
+    ```json tab="JSON result"
+    {
+        "jsonrpc": "2.0",
+        "id": 1,
+        "result": {
+            "0xef1bfb6a12794615c9b0b5a21e6741f01e570185": true,
+            "0x42d4287eac8078828cf5f3486cfe601a275a49a5": true
+        }
+    }
+    ```
+    
+### ibft_getValidatorsByBlockHash
+
+Lists validators for the specified block.
+
+**Parameters**
+
+`data` - 32-byte block hash 
+
+**Returns** 
+
+`result: array of data` - List of validator addresses
+
+!!! example
+    ```bash tab="curl HTTP request"
+    $ curl -X POST --data '{"jsonrpc":"2.0","method":"ibft_getValidatorsByBlockHash","params":["0xbae7d3feafd743343b9a4c578cab5e5d65eb735f6855fb845c00cab356331256"], "id":1}' <JSON-RPC-http-endpoint:port>
+    ```
+    
+    ```bash tab="wscat WS request"
+    {"jsonrpc":"2.0","method":"ibft_getValidatorsByBlockHash","params":["0xbae7d3feafd743343b9a4c578cab5e5d65eb735f6855fb845c00cab356331256"], "id":1}
+    ```
+    
+    ```json tab="JSON result"
+    {
+        "jsonrpc": "2.0",
+        "id": 1,
+        "result": [
+            "0x42d4287eac8078828cf5f3486cfe601a275a49a5",
+            "0xb1b2bc9582d2901afdc579f528a35ca41403fa85",
+            "0xef1bfb6a12794615c9b0b5a21e6741f01e570185"
+        ]
+    }
+    ```
+
+### ibft_getValidatorsByBlockNumber
+
+Lists validators for the specified block. 
+
+**Parameters** 
+
+`quantity|tag` - Integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [Block Parameter](Using-JSON-RPC-API.md#block-parameter). 
```

Comment:
do we really do pending? not confident we would do what is expected with this call ... (i.e. I GUESS it would return the validators in the block being mined .... what happens if you call this on a non-mining node?)

Reply:
We don't do pending. It returns null. The Block Parameter link says it's only used for `eth_getTransactionCount` method